File: Cerebro Utilities/Models/RuleEntity.cs
using Microsoft.Azure.Cosmos.Table;

namespace Cerebro_Utilities.Models
{
    public class RuleEntity : TableEntity
    {
        public const string INDEX_NAME = "cerebrorules-index";

        public const string TABLE_NAME = "CerebroRules";

        public RuleEntity() { }

        public string Description { get; set; }

        public string Footer { get; set; }

        public string Reference { get; set; }

        public string Regex { get; set; }

        public string Terms { get; set; }

        public string Title { get; set; }
    }
}

File: models/ruleEntity.js
const { BaseQualifiedEntity } = require('./baseQualifiedEntity');

class RuleEntity extends BaseQualifiedEntity {
    static DATABASE_SUFFIX = 'rules';
    static COLLECTION = 'rules';

    constructor (document) {
        super(document);

        this.AuthorId = document.AuthorId;
        this.Description = document.Description;
        this.Footer = document.Footer;
        this.Incomplete = document.Incomplete;
        this.Reference = document.Reference;
        this.Regex = document.Regex;
        this.Terms = document.Terms;
        this.Title = document.Title;
        this.Type = document.Type;
    }
}

module.exports = { RuleEntity }